src/ProjectTempoListener.cpp

```diff
@@ -0,0 +1,27 @@
+#include "ProjectTempoListener.h"
+#include "Project.h"
+#include "ProjectTimeSignature.h"
+#include "WaveTrack.h"
+
+ProjectTempoListener::ProjectTempoListener(
+   AudacityProject& project, TrackList& trackList)
+    : mTrackList { trackList }
+    , mTrackListSubstription { trackList.Subscribe(
+         [&project](const TrackListEvent& event) {
+            if (event.mType == TrackListEvent::ADDITION)
+            {
+               const auto tempo = ProjectTimeSignature::Get(project).GetTempo();
+               if (const auto track = event.mpTrack.lock())
+                  track->OnProjectTempoChange(std::nullopt, tempo);
+            }
+         }) }
+{
+}
+
+void ProjectTempoListener::OnProjectTempoChange(
+   const std::optional<double>& oldTempo, double newTempo)
+{
+   for (auto track : mTrackList)
+      if (auto waveTrack = dynamic_cast<WaveTrack*>(track))
+         waveTrack->OnProjectTempoChange(oldTempo, newTempo);
```

Comment:
Again:  I think you can eliminate this class, and even eliminate the std::function alias, and make all of it implementation detail of the Publisher.

And new criticism:  you need to iterate all tracks now, not just Wave tracks.

Reply:
> And new criticism: you need to iterate all tracks now, not just Wave tracks.

Silly me.

Reply:
> And new criticism: you need to iterate all tracks now, not just Wave tracks.

Silly me.